api/test/opentelemetry/trace/tracer_test.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+describe OpenTelemetry::Trace::Tracer do
+  let(:invalid_span) { OpenTelemetry::Trace::Span::INVALID }
+  let(:tracer) { OpenTelemetry::Trace::Tracer.new }
+
+  describe '#current_span' do
+    let(:current_span) { tracer.start_span('current') }
+
+    it 'returns an invalid span by default' do
+      tracer.current_span.must_equal(invalid_span)
+    end
+
+    it 'returns the current span when set' do
+      OpenTelemetry::Context.with(
+        OpenTelemetry::Trace::Tracer.const_get(:CONTEXT_SPAN_KEY),
+        current_span
+      ) do
+        tracer.current_span.must_equal(current_span)
+      end
+    end
+  end
+
+  describe '#in_span' do
+    it 'yields the new span' do
+      tracer.in_span('wrapper') do |span|
+        span.wont_equal(invalid_span)
+        tracer.current_span.must_equal(span)
+      end
+    end
+
+    it 'finishes the new span at the end of the block' do
+      finished = false
+
+      tracer.in_span('wrapper') do |span|
+        # Override the default noop `#finish` for testing
+        span.define_singleton_method(:finish) do
+          finished = true
+        end
+      end
+
+      finished.must_equal(true)
+    end
```

Comment:
I think you could test `in_span` by subclassing `Tracer`, overriding `start_span` to return a mock span, and set expectations on it. Some of the techniques in `batch_span_processor_test.rb` and `simple_span_processor_test.rb` seem useful here.

Reply:
@fbogsany I sure can! I'll see what I can do. Thanks.